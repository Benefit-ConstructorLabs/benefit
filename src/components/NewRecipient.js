import React from 'react';
import PropTypes from 'prop-types';
import RecipientSignUpForm from './RecipientSignUpForm';
// import NewRecipientContainer from '../containers/NewRecipientContainer';

import '../../styles/components/newrecipient.scss';

const NewRecipient = ({
  addRecipient,
  setInputField,
  first_name,
  last_name,
  tel,
  username,
  password,
  photo,
  bio_1,
  bio_2,
  bio_3,
}) => {
  function handleChange(event) {
    setInputField(event.target.name, event.target.value);
  }
  return (
    <React.Fragment>
      <h2>New Recipient</h2>
      <h2 className="newrecipient__title">Start taking digital donations in 3 steps</h2>
      <img src="" alt="" className="newrecipient__header-image" />
      <form
        className="newrecipient__form"
        onSubmit={(event) => {
          event.preventDefault();
          addRecipient();
        }}
      >
        <h2>Add your basic information</h2>
        <ul>
          <li>
            <input
              type="text"
              name="first_name"
              placeholder="First name"
              value={first_name}
              onChange={event => handleChange(event)}
            />
          </li>
          <li>
            <input
              type="text"
              name="last_name"
              placeholder="Last name"
              value={last_name}
              onChange={event => handleChange(event)}
            />
          </li>
          <li>
            <input
              type="text"
              name="tel"
              placeholder="Telephone number"
              value={tel}
              onChange={event => handleChange(event)}
            />
          </li>
          <li>
            <input
              type="text"
              name="username"
              placeholder="Select your username"
              value={username}
              onChange={event => handleChange(event)}
            />
          </li>
          <li>
            <input
              type="password"
              name="password"
              placeholder="Create your password"
              value={password}
              onChange={event => handleChange(event)}
            />
          </li>
          <li>
            <input
              type="text"
              name="photo"
              placeholder="Add a photo"
              value={photo}
              onChange={event => handleChange(event)}
            />
          </li>
        </ul>

        <h2>Now tell people 3 things about you...</h2>
        <ul>
          <li>
            <input
              type="text"
              name="bio_1"
              placeholder="I play..."
              value={bio_1}
              onChange={event => handleChange(event)}
            />
          </li>
          <li>
            <input
              type="text"
              name="bio_2"
              placeholder="I like..."
              value={bio_2}
              onChange={event => handleChange(event)}
            />
          </li>
          <li>
            <input
              type="text"
              name="bio_3"
              placeholder="I enjoy..."
              value={bio_3}
              onChange={event => handleChange(event)}
            />
          </li>
        </ul>
        <button className="newrecipient__btn-submit " type="submit">
          Create account
        </button>
      </form>

      <RecipientSignUpForm />
    </React.Fragment>
  );
};

NewRecipient.propTypes = {
  addRecipient: PropTypes.func.isRequired,
  setInputField: PropTypes.func.isRequired,
  first_name: PropTypes.string.isRequired,
  last_name: PropTypes.string,
  tel: PropTypes.string,
  username: PropTypes.string,
  password: PropTypes.string,
  photo: PropTypes.string,
  bio_1: PropTypes.string,
  bio_2: PropTypes.string,
  bio_3: PropTypes.string,
};

NewRecipient.defaultProps = {
  last_name: '',
  tel: '',
  username: '',
  password: '',
  photo: '',
  bio_1: '',
  bio_2: '',
  bio_3: '',
};
export default NewRecipient;
