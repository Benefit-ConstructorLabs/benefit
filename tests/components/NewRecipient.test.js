import React from 'react';
import renderer from 'react-test-renderer';
import NewRecipient from '../../src/components/NewRecipient';

describe(NewRecipient, () => {
  test.skip('should render correctly', () => {
    const tree = renderer.create(<NewRecipient />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
