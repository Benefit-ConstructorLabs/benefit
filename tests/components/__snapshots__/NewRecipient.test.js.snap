// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewRecipient matches the snapshot 1`] = `
Array [
  <h2>
    New Recipient
  </h2>,
  <h2
    className="newrecipient__title"
  >
    Start taking digital donations in 3 steps
  </h2>,
  <img
    alt=""
    className="newrecipient__header-image"
    src=""
  />,
  <form
    className="neerecipient__form"
    onSubmit={[Function]}
  >
    <h2>
      Add your basic information
    </h2>
    <input
      onChange={[Function]}
      placeholder="First name"
      text="firstname"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Last name"
      text="lastname"
      type="text"
      value={null}
    />
    <input
      onChange={[Function]}
      placeholder="Telephone number"
      text="tel"
      type="text"
      value={null}
    />
    <input
      onChange={[Function]}
      placeholder="Select your username"
      text="username"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Create your password"
      text="password"
      type="password"
    />
    <button
      className="newrecipient__btn-upload-image"
      onClick={[Function]}
      type="submit"
    >
      Upload a picture
    </button>
    <h2>
      Now tell people 3 things about you...
    </h2>
    <input
      onChange={[Function]}
      placeholder="1. I play the piano"
      text="text"
      type="text"
      value={null}
    />
    <input
      onChange={[Function]}
      placeholder="2. I like white coffee"
      text="text"
      type="text"
      value={null}
    />
    <input
      onChange={[Function]}
      placeholder="3. I'm a Rugby fan"
      text="text"
      type="text"
      value={null}
    />
    <button
      className="newrecipient__btn-generate-qr-code "
      onClickUpload={[Function]}
      type="submit"
    >
      Generate Your Unique QR Code
    </button>
  </form>,
  <p>
    Recipient sign-up form
  </p>,
]
`;
